dask/dataframe/shuffle.py

```diff
@@ -437,6 +437,100 @@ def _construct_graph(self):
+    mins = partition_col.map_partitions(M.min)
+    maxes = partition_col.map_partitions(M.max)
+    divisions, sizes, mins, maxes = base.compute(divisions, sizes, mins, maxes)
+    divisions = methods.tolist(divisions)
+    if type(sizes) is not list:
+        sizes = methods.tolist(sizes)
+    mins = methods.tolist(mins)
+    maxes = methods.tolist(maxes)
+
+    empty_dataframe_detected = pd.isnull(divisions).all()
+    if repartition or empty_dataframe_detected:
+        total = sum(sizes)
+        npartitions = max(math.ceil(total / partition_size), 1)
+        npartitions = min(npartitions, df.npartitions)
+        n = len(divisions)
+        try:
+            divisions = np.interp(
+                x=np.linspace(0, n - 1, npartitions + 1),
+                xp=np.linspace(0, n - 1, n),
+                fp=divisions,
+            ).tolist()
+        except (TypeError, ValueError):  # str type
+            indexes = np.linspace(0, n - 1, npartitions + 1).astype(int)
+            divisions = [divisions[i] for i in indexes]
+
+    mins = remove_nans(mins)
+    maxes = remove_nans(maxes)
+    if pd.api.types.is_categorical_dtype(partition_col.dtype):
+        dtype = partition_col.dtype
+        mins = pd.Categorical(mins, dtype=dtype).codes.tolist()
+        maxes = pd.Categorical(maxes, dtype=dtype).codes.tolist()
+
+    return divisions, mins, maxes
+
+
+def sort_values(
+    df,
+    value,
+    npartitions=None,
+    divisions=None,
+    ascending=True,
+    upsample=1.0,
+    partition_size=128e6,
+    **kwargs,
+):
+    """ See _Frame.sort_values for docstring """
+    if not isinstance(value, str):
+        # support ["a"] as input
+        if isinstance(value, list) and len(value) == 1 and isinstance(value[0], str):
+            value = value[0]
+        else:
+            raise NotImplementedError(
+                "Dataframe only supports sorting by a single column which must "
+                "be passed as a string or a list of a sinngle string.\n"
+                "You passed %s" % str(value)
+            )
+    if npartitions == "auto":
+        repartition = True
+        npartitions = max(100, df.npartitions)
+    else:
+        if npartitions is None:
+            npartitions = df.npartitions
+        repartition = False
+
+    sort_by_col = df[value]
+
+    if divisions is None:
+        divisions, mins, maxes = _calculate_divisions(
+            df, sort_by_col, repartition, npartitions, upsample, partition_size
+        )
+
+        if (
+            mins == sorted(mins)
+            and maxes == sorted(maxes)
+            and all(mx < mn for mx, mn in zip(maxes[:-1], mins[1:]))
+            and npartitions == df.npartitions
+        ):
+            # divisions are in the right place
+            divisions = mins + [maxes[-1]]
+            return df.map_partitions(M.sort_values, value)
+    df = rearrange_by_divisions(df, value, divisions)
+    df.divisions = divisions
```

Comment:
I don't think you want to set `df.divisions` to `divisions`. The output of `_calculate_division` is the divisions that would be on the `sort_by_col` if it were the index, but in the case of `sort_by_values` the column does not become the index, so the divisions in this method are not equivalent to the divisions on the resultant dataframe.